clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -246,6 +265,19 @@ public boolean isTransactional() {
         return (attributes() & TRANSACTIONAL_FLAG_MASK) > 0;
     }
 
+    @Override
+    public boolean hasDeleteHorizonMs() {
+        return (attributes() & DELETE_HORIZON_FLAG_MASK) > 0;
+    }
+
+    @Override
+    public long deleteHorizonMs() {
```

Comment:
Maybe we could return `OptionalLong` and get rid of `hasDeleteHorizonMs`?

Reply:
I've removed the `hasDeleteHorizonMs` function from the interface. I've made it so the DefaultRecordBatch still has the function as a private helper, but the logic overall is pretty much unchanged

I've also made the change to return OptionalLong for `deleteHorizonMs` to see how it looks. I think that the current usage in the cases for which this value returns `OptionalLong.Empty` is to just use `RecordBatch.NO_TIMESTAMP`, so I am thinking it's just cleaner to keep it as a `long` and contain the logic for the deleteHorizonMs value or `RecordBatch.NO_TIMESTAMP`